### PR TITLE
weather: add blurred color gradient, changes based on weather

### PR DESCRIPTION
**Pull request by matildepark**
As per mocks.

Colours were indeterminate; I did my best intuitive stab at it with Indigo colours, but the mocks didn't use Indigo colours either. Thus the @urcades review.

Screen Shot 2020-02-18 at 8 55 31 PM

### CODE REVIEW

**Comment by urcades** [comment]
@matildepark :

Screen Shot 2020-02-18 at 11 03 57 PM

Damn, this is nice — will look through the individual color expressions now. Do you have any idea what that space is in the upper left?

**Comment by matildepark** [comment]
> Do you have any idea what that space is in the upper left?

That's the Links tile, which is broken on `os1-rc` pull because the compiled version in the repo is before some tile-specific changes.

**Comment by urcades** [comment]
> > Do you have any idea what that space is in the upper left?
> 
> That's the Links tile, which is broken on `os1-rc` pull because the compiled version in the repo is before some tile-specific changes.

Gotcha, good to know — will leave that out of review

**Comment by matildepark** [comment]
If the gradient is too apparent, feel free to increase the `blur` amount — also let me know if you have a recommendation for the gradient settings ... you should see both of those fairly easily in the inspector, they're inline.

**Comment by urcades** [comment]
Screen Shot 2020-02-18 at 11 15 35 PM

fwiw, gradient rendering on chrome is so, so, so much better. I've been playing around with a few blur options and something around the neighborhood of 80px is looking really good for the overcast tile. 

Currently trying to figure out how to override the weather display and get a sense of what everything else looks like.

**Comment by matildepark** [comment]
Open React DevTools, mouse over Weather component, crack open props -> data -> daily -> icon. Edit it to one of the options in the switch statement and press `Enter`.

image

**Comment by urcades** [comment]
> > Currently trying to figure out how to override the weather display and get a sense of what everything else looks like.
> 
> Open React DevTools, mouse over Weather component, crack open props -> data -> daily -> icon. Edit it to one of the options in the switch statement and press `Enter`.
> 
> image

I'm an idiot: on new machine and didn't have any of that react tooling/toolchain installed ... forgot this stuff isn't built into chrome. Reviewing weather types now.

**Comment by urcades** [comment]
Actually, now that I've posted the above, it looks like your browser renders it fine — I'm using chrome to view this PR, and I can't seem to shake off that blur.

**Comment by matildepark** [comment]
Chrome / Brave:

Screen Shot 2020-02-19 at 1 14 04 PM

**Comment by urcades** [comment]
Holy shit lol, I literally just discovered what the issue is — at varying browser zoom levels, it appears as though that undesired gradient makes an appearance that can't really be controlled:

Kapture 2020-02-19 at 13 11 54

**Comment by urcades** [comment]
I think we can ignore my request for changes

**Comment by matildepark** [comment]
Ah, ok, I can replicate that zoom stuff. That seems out of scope ... mind stamping?

**Comment by urcades** [comment]
@matildepark oh, 80px would be good — in the future do you want me to directly edit PR source with small changes like that? lmk whatever makes your work easier.

**Comment by matildepark** [comment]
Alrighty, ready for @loganallenc stamp. 

> in the future do you want me to directly edit PR source with small changes like that?

It doesn't take much time to amend, don't worry about it.

File: pkg/interface/weather/tile/tile.js [hunk 1]
```diff
@@ -62,18 +46,22 @@ export default class WeatherTile extends Component {
...
+    weatherStyle = { gradient1: "white", gradient2: "white", text: "black" }
+    ) {
     return (
       <div
-        className="pa2 relative bg-white b--black ba"
+        className={"relative b--black ba " + weatherStyle.text}
         style={{
           width: 126,
           height: 126,
+          background: `linear-gradient(135deg, ${weatherStyle.gradient1} 0%,` +
```
Comment (loganallenc) [inline pkg/interface/weather/tile/tile.js]: fascinating. I like it!

File: pkg/interface/weather/tile/tile.js [hunk 2]
```diff
@@ -62,18 +46,22 @@ export default class WeatherTile extends Component {
...
     if (e.keyCode === 13) {
-      e.preventDefault();
-      this.manualLocationSubmit(e.target.value);
+
     }
   }
 
-  renderWrapper(child) {
+  renderWrapper(child,
+    weatherStyle = { gradient1: "white", gradient2: "white", text: "black" }
```
Comment (loganallenc) [inline pkg/interface/weather/tile/tile.js]: weird that you aren't passing these in as separate arguments, as an object is *really* unstructured

File: pkg/interface/weather/tile/tile.js [hunk 3]
```diff
@@ -182,7 +178,91 @@ export default class WeatherTile extends Component {
...
 
     if ('currently' in data && 'daily' in data) {
-      return this.renderWithData(data);
+      let weatherStyle = {
+        gradient1: "",
+        gradient2: "",
+        text: ""
+      };
+
+      switch(data.daily.icon) {
```
Comment (loganallenc) [inline pkg/interface/weather/tile/tile.js]: Put this whole switch statement in its own function. General rule of thumb: one function shouldn't be more than 40 lines

**Comment by matildepark** [comment]
I had some leftover messy code I cleaned up; refactored the style change into its own function; added some comments to help contributors navigate the code. 

I could replicate @urcades' issue myself by just resizing the window around; seemingly [there's a Chrome bug about this](https://stackoverflow.com/questions/57736567/rendering-flickering-glitch-issue-in-chrome-using-css-backdrop-filterblur). 

I still fixed the issue with the black gradient overlay by passing the bordering behaviour to child components (removing it from the paint, since it was taking up 2px of the 126px, thus making the child 124px; placing it in the child stops resizing the backdrop behind the child and thus removing the black gradient overlay).

**Pull request merged**